### PR TITLE
[PLAYER-5644] Geo Blocking error not displayed

### PR DESCRIPTION
**Pull request by oSulyma**
**Ticket goal:** Show correct error message if geoblocking error occured during playback
**How PR achieves the goal:** in category OOOoyalaPlayer (StateMachine) setter for state used to send wrong (previouse) state with OOOoyalaPlayerStateChangedNotification. 
OOSkinPlayerObserver switched to read state from  notification's user info instead from player directly.

**Affected repo-s:** iOS-SDK + native-Skin-SDK + Skin Config

**Unit tests:** Unit test not added.
**SampleApp for testing**: OOyalaSkin Sample app. 
Asset and other: Please see https://git.corp.ooyala.com/projects/PBA/repos/ios-sdk/pull-requests/561/overview

### CODE REVIEW

File: sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m [hunk 1]
```diff
@@ -347,6 +347,12 @@ - (void)bridgeCurrentItemChangedNotification:(NSNotification *)notification {
 - (void)bridgeStateChangedNotification:(NSNotification *)notification {
   NSString *stateString = [OOOoyalaPlayerStateConverter playerStateToString:self.player.state];
   
+  NSNumber *stateObject = notification.userInfo[@"newState"];
+  if (stateObject) {
```
Comment (maxkupetskii) [inline sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: I think this should be default behavior, i.e.:
```suggestion
NSNumber *stateObject = notification.userInfo[@"newState"] ?: self.player.state;
NSString *stateString = [OOOoyalaPlayerStateConverter playerStateToString:stateObject];
```

**Review by maxkupetskii — CHANGES_REQUESTED** [review]

File: sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m (hunk 1 above)
Comment (oSulyma) [reply sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: Agree, if it is okey that if notification.userInfo[@"newState"]  is nil (in unknown reason) observer will get string for initial state in enum: OOOoyalaPlayerStateInit

**Review by loginov-rocks — APPROVED** [review]

File: sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m (hunk 1 above)
Comment (oSulyma) [reply sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: Can't do it in this way.
self.player.state - it's NSUInteger, but  notification.userInfo[@"newState"] it's NSNumber.
But anyway I simplified this snippet.

Comment (oSulyma) [reply sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: @maxkupetskii  Please, review it again

Comment (maxkupetskii) [reply sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: Use @(self.player.state) in my snippet

Comment (oSulyma) [reply sdk/iOS/OoyalaSkinSDK/OOSkinPlayerObserver.m]: Done

**Review by maxkupetskii — APPROVED** [review]

**Review by DmitriiMaslov — APPROVED** [review]

**Pull request merged**